sdks/python/apache_beam/pvalue.py

```diff
@@ -208,9 +233,23 @@ def from_runner_api(proto, context):
     # deserialization.  It will be populated soon after this call, in
     # Pipeline.from_runner_api(). This brief period is the only time that
     # PCollection.pipeline is allowed to be None.
+
+    try:
+      element_type = context.element_type_from_coder_id(proto.coder_id)
```

Comment:
Have this method do the ExternalCoder stuff rather than catching an exception here. Possibly it would not even be needed if the coders map is properly populated beforehand.